packages/ses/src/eval-scope.js

```diff
@@ -1,6 +1,29 @@
-import { FERAL_EVAL, create, freeze, defineProperties } from './commons.js';
+import { FERAL_EVAL, create, defineProperties, freeze } from './commons.js';
+
+import { assert } from './error/assert.js';
+
+const { details: d } = assert;
+
+export const makeEvalScopeKit = () => {
+  // In makeSafeEvaluator, we ensure that we consume at least as many stack
+  // frames when granting access to FERAL_EVAL than we consume when we are
+  // immediately withdrawing access, such that the safe evaluator can reach it
+  // in the lexical scope of the confined program once (to evaluate the
+  // confined code) and confined code cannot access it thereafter.
+  // We do this to frustrate stack-bumping attacks, where the caller arranges
+  // for the stack to be a number of frames short of inducing a `RangeError` in
+  // order to interrupt a program that may not be able to clean up safely.
+  // In this case, there is a plausible but very narrow window of opportunity
+  // between defining `oneTimeEvalProperties` on the `evalScope` and when the
+  // confined program reaches for the eval property on the `evalScope` through
+  // a lexical reference where a program could induce a RangeError that would
+  // prevent the program from reaching `delete evalScope.eval` either here or
+  // in the fallback position in `makeSafeEvaluator` where we attempt to delete
```

Comment:
```suggestion
  // in the fallback position in `makeSafeEvaluator`, where we attempt to delete
```
How long is that sentence? ;)